src/target/target_flash.c

```diff
@@ -312,3 +312,65 @@ bool target_flash_complete(target_s *target)
 	target_exit_flash_mode(target);
 	return result;
 }
+
+static bool flash_blank_check(target_flash_s *flash, target_addr_t src, size_t len, target_addr_t *mismatch)
+{
+	bool result = true; /* Catch false returns with &= */
+	target_s *target = flash->t;
+	platform_timeout_s timeout;
+	platform_timeout_set(&timeout, 500);
+
+	while (len) {
```

Comment:
That was for `const size_t local_len = MIN(flash->writebufsize - offset, len);` in case a flash bank capacity is not an even/integer multiple of `writebufsize`. If there is such a guarantee, then I think I can simplify into `for`-loop.

Reply:
That is a guarantee you can depend on. Even if it wasn't, `MIN(len - offset, flash->writebufsize)` for local_len would still give you the right answer with the suggested restructure.

Reply:
How about this form then? 8c8db531

Reply:
That looks considerably better!